### PR TITLE
feat(deps): react@18 support

### PR DESCRIPTION
**Pull request by chanceaclark**
## What?

🎉  Finally... the time has come... React@18 support! 🎉 

## Why?

All the necessary libraries have finally been updated to support React@18.

### Documentation site working as expected
screencapture-localhost-3000-2022-09-20-12_39_59

## Testing/Proof

Tested in channel-manager within the Control Panel.

### CODE REVIEW

File: packages/big-design/src/components/Select/spec.tsx [hunk 1]
```diff
@@ -546,10 +547,9 @@ test('closing the Select triggers onClose', async () => {
 
   const button = await screen.findByLabelText('toggle menu');
 
-  await act(async () => {
-    await fireEvent.click(button);
-    await fireEvent.click(button);
-  });
+  await userEvent.click(button);
+
+  await userEvent.click(button);
```
Comment (chanceaclark) [inline packages/big-design/src/components/Select/spec.tsx:552]: Test was failing with `act` so switched over to `userEvent`.

File: packages/big-design/src/components/Worksheet/hooks/useUpdateItems/spec.tsx [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
+  renderHook(() => useUpdateItems(), {
+    wrapper: class Wrapper extends Component<PropsWithChildren<unknown>> {
+      override componentDidCatch(err: unknown) {
+        error = err;
+      }
+      override render() {
+        return this.props.children;
+      }
+    },
+  });
```
Comment (chanceaclark) [inline packages/big-design/src/components/Worksheet/hooks/useUpdateItems/spec.tsx:25]: When they ported over `renderHook` to v13 of `@testing-library/react` they shipped it with a leaner API. Now we don't get errors with this, but according to https://github.com/testing-library/react-testing-library/pull/991#issuecomment-1207138334 this is the best way to achieve the same thing.

File: packages/docs/components/ColorCards/ColorCard.tsx [hunk 3]
```diff
@@ -58,7 +58,9 @@ const ColorCard: React.FC<ColorCardProps> = ({ colorCard }) => {
           {description.text}{' '}
           {Boolean(description.href) && (
             <Link href={description.href} target="_blank">
-              <Small color="inherit">Read more</Small>
+              <Small as="span" color="inherit">
+                Read more
+              </Small>
```
Comment (chanceaclark) [inline packages/docs/components/ColorCards/ColorCard.tsx:63]: Screen Shot 2022-09-20 at 12 28 14

File: packages/docs/PropTables/AlertPropTable.tsx [hunk 4]
```diff
@@ -18,8 +18,11 @@ const alertProps: Prop[] = [
     defaultValue: 'false',
     description: (
       <>
-        <Text>Auto dismiss after 5 seconds.</Text>
-        <Small>Note: Only valid when used with AlertManager.</Small>
+        Auto dismiss after 5 seconds.
+        {/* I hate using a br but the as prop doesn't support div for now */}
+        {/* TODO: Support div for as prop */}
+        <br />
+        <Small as="span">Note: Only valid when used with AlertManager.</Small>
```
Comment (chanceaclark) [inline packages/docs/PropTables/AlertPropTable.tsx:25]: Screen Shot 2022-09-20 at 12 36 52

File: packages/docs/PropTables/BoxPropTable.tsx [hunk 5]
```diff
@@ -64,8 +64,11 @@ const boxProps: Prop[] = [
     types: ['sticky', 'fixed', 'modalBackdrop', 'modal', 'tooltip', 'popover'],
     description: (
       <>
-        <Text>Determines type of z-index to be applied.</Text>
-        <Small>Types are in order (sticky is low, popover is high).</Small>
+        Determines type of z-index to be applied.
+        {/* I hate using a br but the as prop doesn't support div for now */}
+        {/* TODO: Support div for as prop */}
+        <br />
+        <Small as="span">Types are in order (sticky is low, popover is high).</Small>
```
Comment (chanceaclark) [inline packages/docs/PropTables/BoxPropTable.tsx:71]: Screen Shot 2022-09-20 at 12 38 51

File: packages/docs/components/CodePreview/CodePreview.tsx [hunk 6]
```diff
@@ -61,7 +63,7 @@ function transformCode(input: string): string {
 }
 
 export interface CodePreviewProps {
-  children?: React.ReactNode;
+  children?: CodePreviewChildren;
```
Comment (jorgemoya) [inline packages/docs/components/CodePreview/CodePreview.tsx:66]: 👏

File: packages/big-design/src/components/Worksheet/hooks/useUpdateItems/spec.tsx (hunk 2 above)
Comment (jorgemoya) [reply packages/big-design/src/components/Worksheet/hooks/useUpdateItems/spec.tsx:25]: 😵‍💫

File: packages/docs/PropTables/AlertPropTable.tsx (hunk 4 above)
Comment (jorgemoya) [reply packages/docs/PropTables/AlertPropTable.tsx:25]: :cringe:

**Review by jorgemoya — APPROVED** [review]
Thanks for working on this! 👍

**Review by bc-juanvasquez — APPROVED** [review]
💭

**Comment by chanceaclark** [comment]
Going to wait till https://github.com/bigcommerce/big-design/pull/985 is released before releasing this PR.

**Review by MariaJose — APPROVED** [review]
🥳

**Pull request merged**